# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/mac/DocAccessibleWrap.mm
# Commit: 0441d1553b6b
# Full Hash: 0441d1553b6bbf99d711300680eccd335e0b3545
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-07-03 03:56:55
# Regressor Bug: 1649575
# File Overlap Count: 1
# Description:
#   Bug 1649575 - Part 3: Introduce MOXTextMarkerDelegate. r=morgan
#   
#   The delegate is associated with accessible documents (either local or remote).
#   
#   I made a separate protocol for all the text marker stuff as it really is a seperate API.
# ==============================================================================

diff -r 65f8e705d730 -r 0441d1553b6b accessible/mac/DocAccessibleWrap.mm
--- a/accessible/mac/DocAccessibleWrap.mm	Thu Jul 02 21:24:39 2020 +0000
+++ b/accessible/mac/DocAccessibleWrap.mm	Thu Jul 02 21:33:07 2020 +0000
@@ -1,4 +1,4 @@
-/* -*- Mode: C; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+/* -*- Mode: Objective-C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
@@ -6,6 +6,7 @@
 #include "DocAccessibleWrap.h"
 
 #import "mozAccessible.h"
+#import "MOXTextMarkerDelegate.h"
 
 using namespace mozilla;
 using namespace mozilla::a11y;
@@ -13,4 +14,9 @@
 DocAccessibleWrap::DocAccessibleWrap(dom::Document* aDocument, PresShell* aPresShell)
     : DocAccessible(aDocument, aPresShell) {}
 
+void DocAccessibleWrap::Shutdown() {
+  [MOXTextMarkerDelegate destroyForDoc:this];
+  DocAccessible::Shutdown();
+}
+
 DocAccessibleWrap::~DocAccessibleWrap() {}